tests/ship_restart_test.py

```diff
@@ -0,0 +1,304 @@
+#!/usr/bin/env python3
+
+import os
+import tempfile
+import shutil
+import signal
+
+from TestHarness import Cluster, TestHelper, Utils, WalletMgr
+
+###############################################################################
+# ship_restart_test
+# 
+# This test verifies SHiP shuts down gracefully or recovers when restarting
+# with various scenarios of corrupted log and/or index files.
+# 
+###############################################################################
+
+Print=Utils.Print
+
+args = TestHelper.parse_args({"--dump-error-details","--keep-logs","-v","--leave-running","--unshared"})
+
+Utils.Debug=args.v
+cluster=Cluster(unshared=args.unshared, keepRunning=args.leave_running, keepLogs=args.keep_logs)
+dumpErrorDetails=args.dump_error_details
+walletPort=TestHelper.DEFAULT_WALLET_PORT
+
+totalProducerNodes=1
+totalNonProducerNodes=1 # for SHiP node
+totalNodes=totalProducerNodes+totalNonProducerNodes
+
+walletMgr=WalletMgr(True, port=walletPort)
+testSuccessful=False
+
+prodNodeId = 0
+shipNodeId = 1
+
+origStateHistoryLog   = ""
+stateHistoryLog       = ""
+origStateHistoryIndex = ""
+stateHistoryIndex     = ""
+
+# Returns True if file1 and file2 contain the same content
+def equalFiles(file1, file2):
```

Comment:
Can you use `Utils.compareFiles` instead?

Reply:
Thanks. I did not notice that function. I just need to add an extra argument for file mode (the original version does not work for binary file comparison).